libsolidity/analysis/TypeChecker.cpp

```diff
@@ -955,6 +955,16 @@ void TypeChecker::endVisit(Return const& _return)
 	}
 }
 
+void TypeChecker::endVisit(EmitStatement const& _emit)
+{
+	if (
+		_emit.eventCall().annotation().kind != FunctionCallKind::FunctionCall ||
+		dynamic_cast<FunctionType const&>(*type(_emit.eventCall().expression())).kind() != FunctionType::Kind::Event
+	)
+		m_errorReporter.typeError(_emit.eventCall().expression().location(), "Expression has to be an event.");
```

Comment:
You are using "event call" elsewhere, perhaps worth using the same for consistency.

Reply:
Or how about going with "event invocation" in all places?